webservices/ruby/style_guide.md

```diff
@@ -0,0 +1,492 @@
+      x + y
+    end
+
+    # good
+    def sum(operand_1, operand_2)
+      x = operand_1
+      y = operand_2
+      x + y
+    end
+  ```
+
+### Prefer && and || over AND and OR.
+  ```ruby
+    # bad
+    def valid_pizza?
+      heavy_pizza? or (light_pizza? and big_pizza?)
+    end
+
+    # good
+    def valid_pizza?
+      heavy_pizza? || (light_pizza? && big_pizza?)
+    end
+  ```
+
+### Prefer ! over not.
+  ```ruby
+    # bad
+    def valid_pizza?
+      not small_pizza?
+    end
+
+    # good
+    def valid_pizza?
+      !small_pizza?
+    end
+  ```
+
+### Use %{} for single-line strings needing interpolation and double-quotes.
+  ```ruby
+    # bad
+    def insert_annoying_thing_into_string
+      thing = "annoying"
+      'i am an "' + thing + '" string'
+    end
+
+    # good
+    def insert_annoying_thing_into_string
+      thing = "annoying"
+      %{i am an "#{thing}" string}
+    end
+  ```
+
+### Avoid using semicolons.
+```ruby
+  # bad
+  def sum(operand_1, operand_2)
+    operand_1 + operand_2;
+  end
+
+  # good
+  def sum(operand_1, operand_2)
+    operand_1 + operand_2
+  end
+```
+
+### Avoid organizational comments.
+  ```ruby
+    # bad
+    class Dog
+      # associations
+      belongs_to :user
+      belongs_to :home
+      has_many :favorite_foods
+
+      # validations
+      validates :name, presence: true
+      validates :color, presence: true
+      validates :breed, presence: true
+    end
+
+    # good
+    class Dog
+      belongs_to :user
+      belongs_to :home
+      has_many :favorite_foods
+
+      validates :name, presence: true
+      validates :color, presence: true
+      validates :breed, presence: true
+    end
+  ```
+
+### Use CamelCase for classes and modules, snake_case for variables, methods, and file names, SCREAMING_SNAKE_CASE for constants.
+  ```ruby
+    # bad
+    # app/models/LoudCamelSnakes.rb
+    class loudCamelSnakes
+      maximum_camelsnake_capacity = 450
+
+      def SneakyCamelSnake
+      end
+    end
+
+    # good
+    # app/models/loud_camel_snakes.rb
+    class LoudCamelSnakes
+      MAXIMUM_CAMELSNAKE_CAPACITY = 450
+
+      def sneaky_camel_snake
+      end
+    end
+  ```
+
+# Conditionals
+
+### Avoid lines that end with conditionals.
+  ```ruby
+    # bad
+    def sum(operand_1, operand_2)
+      operand_1 + operand_2 if valid_operands?
+    end
+
+    # good
+    def sum(operand_1, operand_2)
+      if valid_operands?
+        operand_1 + operand_2
+      end
+    end
+  ```
+
+### Avoid ternary operators.
+  ```ruby
+    # bad
+    def sum(operand_1, operand_2)
+      valid_operands? ? operand_1 + operand_2 : raise InvalidOperandError
+    end
+
+    # good
+    def sum(operand_1, operand_2)
+      if valid_operands?
+        operand_1 + operand_2
+      else
+        raise InvalidOperandError
+      end
+    end
+  ```
+
+### Prefer if over unless.
+  - use `unless` for a single negation; use `if` otherwise.
+  ```ruby
+  # bad
+    def best_pizza?
```

Comment:
I don't like these examples.  Could they maybe be changed?  The problem here is that you have a function call that returns a value if the conditional is met but no value otherwise.  It would be interpreted as `falsey` so there would probably not be a bug, but my best advice would be to leave out the if and return and just use the conditional.  So, 
```def best_pizza?
  ! small_pizza?
end```
That way, you get an answer in either case.

Reply:
Yeah, you're right; these examples are the worst.  I made them (hopefully) less terrible.